distro/config.go

```diff
@@ -62,6 +138,148 @@ func (c config) Validate() error {
 	return nil
 }
 
+type nonePropagatorType struct{ propagation.TextMapPropagator }
+
+var nonePropagator = nonePropagatorType{}
+
+// propagators maps environment variable values to TextMapPropagator creation
+// functions.
+var propagators = map[string]func() propagation.TextMapPropagator{
+	// W3C Trace Context.
+	"tracecontext": func() propagation.TextMapPropagator {
+		return propagation.TraceContext{}
+	},
+	// W3C Baggage
+	"baggage": func() propagation.TextMapPropagator {
+		return propagation.Baggage{}
+	},
+	// B3 Single
+	"b3": func() propagation.TextMapPropagator {
+		return b3.New(b3.WithInjectEncoding(b3.B3SingleHeader))
+	},
+	// B3 Multi
+	"b3multi": func() propagation.TextMapPropagator {
+		return b3.New(b3.WithInjectEncoding(b3.B3MultipleHeader))
+	},
+	// Jaeger
+	"jaeger": func() propagation.TextMapPropagator {
+		return jaeger.Jaeger{}
+	},
+	// AWS X-Ray.
+	"xray": func() propagation.TextMapPropagator {
+		return xray.Propagator{}
+	},
+	// OpenTracing Trace
+	"ottrace": func() propagation.TextMapPropagator {
+		return ot.OT{}
+	},
+	// None, explicitly do not set a global propagator.
+	"none": func() propagation.TextMapPropagator {
+		return nonePropagator
```

Comment:
nit: maybe just use `nil` instead? having `nonePropagatorType` gives a false impression that the interface is implemented. It would solve "my problems" that `nonePropagator` is not a "NullObject" having a no-op implementation. `nil` is nothing - very obvious 😄 feel free to resolve if you disagree or do not want to lose time on it